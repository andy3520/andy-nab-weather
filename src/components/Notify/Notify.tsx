import React from 'react';

export interface INotifyProps {
  type: 'info' | 'error' | undefined;
  msg: string;
}

export const NOTIFY_CLASS_NAME = {
  info: 'bg-indigo-200 text-indigo-600',
  success: 'bg-green-200 text-green-600',
  error: 'bg-red-200 text-red-600',
};

const Notify: React.FC<INotifyProps> = ({ type, msg }) => {
  const typeClass = !type ? NOTIFY_CLASS_NAME.info : NOTIFY_CLASS_NAME[type];

  return (
    <div className={`${typeClass} shadow px-4 py-5 sm:px-6 rounded-lg`}>
      {msg}
    </div>
  );
};

export default Notify;
