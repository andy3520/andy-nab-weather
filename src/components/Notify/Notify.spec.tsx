import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import Notify, { INotifyProps, NOTIFY_CLASS_NAME } from './Notify';

describe('Notify', () => {
  test('should match snapshot', () => {
    const type = 'info';
    const msg = 'Officia sunt enim occaecat nostrud do commodo.';
    const { asFragment } = render(<Notify type={type} msg={msg} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test.each`
    type         | msg
    ${'info'}    | ${'Est ex reprehenderit consequat.'}
    ${'error'}   | ${'Laborum nisi aliqua sit.'}
    ${''}        | ${'Ut eu voluptate qui nisi nulla.'}
    ${undefined} | ${'Adipisicing officia aliquip duis.'}
  `(
    'should render notify style for type "$type" and message "$msg" correctly',
    ({ type, msg }: INotifyProps) => {
      const { container } = render(<Notify type={type} msg={msg} />);
      const typeClass = !type
        ? NOTIFY_CLASS_NAME.info
        : NOTIFY_CLASS_NAME[type];

      expect(container.firstChild).toHaveClass(typeClass);
      expect(container.firstChild).toHaveTextContent(msg);
    }
  );
});
