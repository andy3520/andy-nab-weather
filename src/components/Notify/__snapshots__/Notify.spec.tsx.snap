// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notify should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-indigo-200 text-indigo-600 shadow px-4 py-5 sm:px-6 rounded-lg"
  >
    Officia sunt enim occaecat nostrud do commodo.
  </div>
</DocumentFragment>
`;
